### PR TITLE
Additional records

### PR DESCRIPTION
**Pull request by chbatey**
Also includes commits from https://github.com/akka/akka-management/pull/274#pullrequestreview-148442323

### CODE REVIEW

**Comment by chbatey** [comment]
2.5.15 released now :)

File: discovery-dns/src/main/scala/akka/discovery/dns/DnsSimpleServiceDiscovery.scala [hunk 1]
```diff
@@ -3,25 +3,60 @@
...
  * Looks for A records for a given service.
  */
 class DnsSimpleServiceDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {
 
-  // Required for async dns
-  AkkaVersion.require("discovery-dns", "2.5.14")
+  import DnsSimpleServiceDiscovery._
+
+  // Required for async dns, 15 for additional records
+  AkkaVersion.require("discovery-dns", "2.5.15")
```
Comment (ktoso) [inline discovery-dns/src/main/scala/akka/discovery/dns/DnsSimpleServiceDiscovery.scala]: 👍

File: discovery-dns/src/main/scala/akka/discovery/dns/DnsSimpleServiceDiscovery.scala [hunk 2]
```diff
@@ -3,25 +3,60 @@
...
+        } else {
+          addresses
+        }
+      case other => im.Seq.empty[ResolvedTarget]
+    }
+
+    Resolved(srvRequest, addresses)
+  }
+
+}
```
Comment (ktoso) [inline discovery-dns/src/main/scala/akka/discovery/dns/DnsSimpleServiceDiscovery.scala]: 👍

File: project/Dependencies.scala [hunk 3]
```diff
@@ -3,9 +3,8 @@ import Keys._
 
 object Dependencies {
 
-  val AkkaVersion = "2.5.14"
-  // TODO #281 update to 10.1.x when play and lagom are released
-  val AkkaHttpVersion = "10.0.13"
+  val AkkaVersion = "2.5.15"
+  val AkkaHttpVersion = "10.0.13" // TODO update to 10.1.x as soon as possible
```
Comment (ktoso) [inline project/Dependencies.scala]: please keep the other note: `   // TODO #281 update to 10.1.x when play and lagom are released `

Comment (chbatey) [reply project/Dependencies.scala]: Merge mistake, will add back

File: discovery-dns/src/main/scala/akka/discovery/dns/DnsSimpleServiceDiscovery.scala [hunk 4]
```diff
@@ -3,25 +3,44 @@
...
-import akka.io.dns.{ AAAARecord, ARecord, DnsProtocol, SRVRecord }
 import akka.io.dns.DnsProtocol.{ Ip, Srv }
+import akka.io.dns.{ AAAARecord, ARecord, DnsProtocol, SRVRecord }
+
+object DnsSimpleServiceDiscovery {
+  def srvRecordsToResolved(srvRequest: String, resolved: DnsProtocol.Resolved): Resolved = {
+    val ips: Map[String, InetAddress] = resolved.additionalRecords.collect {
+      case a: ARecord => a.name -> a.ip
+      case aaaa: AAAARecord => aaaa.name -> aaaa.ip
+    }.toMap
```
[the remark this answers is not in the document]
Comment (raboof) [reply discovery-dns/src/main/scala/akka/discovery/dns/DnsSimpleServiceDiscovery.scala]: Seems elegant :+1: - so this way it's up to the consumer of the API what to do when the IP address is `None`. Makes sense.

[the remark this answers is not in the document]
Comment (chbatey) [reply discovery-dns/src/main/scala/akka/discovery/dns/DnsSimpleServiceDiscovery.scala]: Yeah which could be to resolve the target some other way. Which is what most db/http clients will do if they pass that into them

**Pull request merged**